src/utils.js

```diff
@@ -672,6 +675,35 @@ function getCompileFn(implementation, options) {
       };
     }
 
+    if (options.api === "modern-compiler") {
+      return async (sassOptions) => {
+        // eslint-disable-next-line no-underscore-dangle
+        const webpackCompiler = loaderContext._compiler;
+        const { data, ...rest } = sassOptions;
+
+        // Some people can run the loader in a multi-threading way;
+        // there is no webpack compiler object in such case.
+        if (webpackCompiler) {
+          if (!sassModernCompilers.has(implementation)) {
+            // Create a long-running compiler process that can be reused
+            // for compiling individual files.
+            const compiler = await implementation.initAsyncCompiler();
+            // Check again because awaiting the initialization function
+            // introduces a race condition.
+            if (!sassModernCompilers.has(implementation)) {
+              sassModernCompilers.set(implementation, compiler);
+              webpackCompiler.hooks.shutdown.tap("sass-loader", () => {
+                compiler.dispose();
+              });
+            }
+          }
+          return sassModernCompilers.get(implementation).compileStringAsync(data, rest);
```

Comment:
Let's add a test case where we have two sass loaders with different options and use `modern-compiler`, just make sure everything works fine

Reply:
Does the test from https://github.com/webpack-contrib/sass-loader/pull/1195/commits/183765a1d26d9c16523bb35158cbb2449e280444 look good to you? I'm not sure how to otherwise have two sass-loaders; we can't chain them within a single webpack config, right?

Reply:
I am fine with test, maybe I will add more later, want to look at this today